unlock_indicator.c

```diff
@@ -32,10 +32,20 @@ extern double ring_width;
 #define BUTTON_RADIUS (circle_radius)
 #define RING_WIDTH (ring_width)
 #define BUTTON_SPACE (BUTTON_RADIUS + (RING_WIDTH / 2))
-#define BUTTON_CENTER (BUTTON_RADIUS + (RING_WIDTH / 2))
 #define BUTTON_DIAMETER (2 * BUTTON_SPACE)
-#define CLOCK_WIDTH 400
-#define CLOCK_HEIGHT 200
+
+typedef struct {
```

Comment:
This struct should be in the header file

Reply:
Well, that, or there should be an additional .c/.h file for this stuff if we break it out (which is probably best, but there's so many globals in this file that it'll probably end up being a mess)

Reply:
I included it in the c file as there is no need to make this struct public / let users of the header parse this as it is not an interface struct but a "private" one.

Reply:
I still put things like the rgb structs into the header file, mostly just to keep hings clean.